blueocean-dashboard/src/main/js/components/Capability.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react';
+import update from 'react-addons-update';
+import { classMetadataStore } from '@jenkins-cd/js-extensions';
+
+export const capabilityStore = classes => ComposedComponent => class extends Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            capabilities: {},
+        };
+    }
+
+    componentDidMount() {
+        const self = this;
+        let _classes = classes(this.props);
+
+        if (typeof _classes === 'string') {
+            _classes = [_classes];
+        }
+
+        for (const _class of _classes) {
+            classMetadataStore.getClassMetadata(_class, (classMeta) => {
+                self._setState(_class, self._classesToObj(classMeta.classes));
+            });
+        }
+    }
+
+    componentWillUnmount() {
+        this.unmounted = true;
+    }
+    
+    _classesToObj(_classes) {
+        if (!_classes) {
+            return {
+                has: () => false,
+            };
+        }
+
+        return {
+            _classes,
+            has: capability => _classes.find(_class => _class === capability) !== undefined,
+        };
+    }
+    _setState(key, value) {
+        // Block calls to setState for components that are
+        // not in a mounted state.
+        if (!this.unmounted) {
+            const newData = { capabilities: {} };
+            newData.capabilities[key] = { $set: value };
+            this.setState(previousState => update(previousState, newData));
```

Comment:
`previousState`? I don't see it defined anywhere. Maybe some comments here as to what's happening. No obvious to me anyway.